### ISSUE TITLE
[really good idea] Deprecate Fibers, encourage async/await.

### ISSUE THREAD

**Issue by trusktr** [meteor/meteor#5631]
I think that'll solve the problem of letting developers write code in a sync fashion, but without encouraging them to write absolutely everything in a sync fashion and therefore lose out on the benefits of async.

I feel that with Fibers it's too easy for someone to write everything in a 100% sync fashion, which defeats the purpose of JavaScript's async. Now, you may say "everything you execute with Fibers runs async" and that may be true, but if a developer write everything in a sync fashion, then what's happening is the equivalent of always using [`async.series`](https://github.com/caolan/async#seriestasks-callback), and so nothing will ever run in parallel, despite that it is async and that it *can* run in parallel.

I think Fibers encourages antipatterns that are really easy to follow and shoot yourself in the foot with.

Now, I'm not arguing that Fibers isn't powerful. In fact, it can be really powerful when used properly, but the learning curve to *actually use fibers properly* is quite high, and that aspect is hidden behind Meteor's user-friendly APIs. [This article](https://meteorhacks.com/fibers-eventloop-and-meteor) even says

> Fibers is really good if you use it correctly (Meteor does it well).

Noting the "if you use it correcly" which I think is hidden behind the Meteor API, and trickier to learn than async/await.

Now, imagine how much better all of this would be if Meteor taught developers how to program with async/await. :blush: 

Of course, it probably wasn't possible to even bring async/await into the picture, but now that @benjamn is working on https://github.com/meteor/meteor/pull/5475, it will soon be possible. :+1:

**Comment by benjamn** [comment]
Not sure if you've seen this talk I gave earlier this year about my vision for the future of Fibers/coroutines in Meteor, but it sounds like you may find it interesting: http://benjamn.github.io/goto2015-talk/

**tl;dr**, Fibers are an excellent way of implementing `async` functions on the server (where Fibers are available), and they also leave open the exciting possibility of `await` expressions that can appear outside of `async` functions.

**Comment by mitar** [comment]
-1 on deprecating Fibers. I really like that there is a stack supporting them officially.

**Comment by rozzzly** [comment]
Idk how this would be possible, and from what the gentleman above point out, there would several major implications in such a proposal... Just playing devil's advocate here... But, what about _polyfilling_ `async/wait`? Pretty sure the ecmascript spec is not conducive to what fibers can offer, but I'll throw this possibility our there regardless. :p

**Comment by trusktr** [comment]
@benjamn I like that await can implemented via Fibers, that's cool! Is there an advantage to the `await()` calls via Fiber over the await/async syntax (besides not being required to call `await()` inside an async function)?

**Comment by trusktr** [comment]
I found this, built with fibers: https://github.com/yortus/asyncawait

**Comment by stubailo** [comment]
Duplicate of https://github.com/meteor/meteor/issues/5460
